### PR TITLE
Cleanup. 

### PR DESCRIPTION
**Pull request by zeeshanlakhani**
Know this seems like a ton, but it's mostly deletion and renaming, with very little of anything new. 

I broke each related parts into commits, please use that for separation.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,35 +1,23 @@
...
-        # "test/tcp-reconstruction",
-        # "test/acl-fw",
-        # "test/embedded-scheduler-test",
-        # "test/embedded-scheduler-dependency-test"
+        "examples/echo-reply",
+        "examples/ipv4or6",
+        "examples/macswap",
+        "examples/mtu-too-big",
+        "examples/op-errors",
+        # "examples/srv6-sighup-flow",
```
Comment (zeeshanlakhani) [inline Cargo.toml]: On my next TODO.

File: README.md [hunk 2]
```diff
@@ -61,7 +61,7 @@ VM settings within the `Vagrantfile` once it has been symlinked.
 8. After step 6, you'll be in the container and then can compile and test NetBricks via
```
Comment (zeeshanlakhani) [inline README.md]: Note: I will tackle a more complete README as well (and link to old netbricks version) in my next PR with more tests.

File: examples/op-errors/src/main.rs [hunk 3]
```diff
@@ -47,14 +50,40 @@ where
...
     let pipelines: Vec<_> = ports
         .iter()
-        .map(|port| nf(ReceiveBatch::new(port.clone())).send(port.clone()))
+        .map(|port| {
+            ReceiveBatch::new(port.clone())
+                .map(|p| p.parse::<Ethernet>())
+                .filter(|p| match p.ether_type() {
+                    EtherTypes::Ipv6 => true,
+                    _ => false,
+                })
```
Comment (drunkirishcoder) [inline examples/op-errors/src/main.rs]: you can simplify this to `p.ether_type() == EtherTypes::Ipv6`.

File: examples/op-errors/src/main.rs [hunk 4]
```diff
@@ -47,14 +50,40 @@ where
...
     println!("Running {} pipelines", pipelines.len());
     for pipeline in pipelines {
         sched.add_task(pipeline).unwrap();
     }
 }
 
+fn throw_mama_from_the_train(packet: Ipv6) -> Result<Ipv6> {
+    let hextets = packet.src().segments();
+    let prefix = Ipv6Addr::new(hextets[0], hextets[1], hextets[2], hextets[3], 0, 0, 0, 0);
+    if prefix == Ipv6Addr::from_str("da75::").unwrap() {
```
Comment (drunkirishcoder) [inline examples/op-errors/src/main.rs]: use `cidr` match here?

File: framework/src/config/mod.rs [hunk 5]
```diff
@@ -1,9 +1,22 @@
+use failure::Fail;
```
Comment (drunkirishcoder) [inline framework/src/config/mod.rs]: We should put it on our `todo` to rip this stuff out and use [clap](https://github.com/clap-rs/clap) instead and integrate this with the config-rs structures we already using. Low priority though.

File: framework/src/packets/ip/mod.rs [hunk 6]
```diff
@@ -115,21 +116,51 @@ impl Flow {
...
+
     pub fn protocol(&self) -> ProtocolNumber {
         self.protocol
     }
+
+    pub fn set_protocol(&mut self, protocol: ProtocolNumber) {
+        self.protocol = protocol
+    }
+
+    pub fn reverse_flow(&self) -> Self {
```
Comment (drunkirishcoder) [inline framework/src/packets/ip/mod.rs]: maybe just call it `reverse` since you are already in the struct.

File: framework/src/packets/tcp.rs [hunk 7]
```diff
@@ -390,6 +390,15 @@ impl<E: IpPacket> Tcp<E> {
...
     }
 
+    #[inline]
+    pub fn stamp_flow(&mut self, flow: Flow) -> Result<()> {
+        self.envelope_mut().set_src(flow.src_ip())?;
+        self.envelope_mut().set_dst(flow.dst_ip())?;
+        self.set_src_port(flow.src_port());
+        self.set_dst_port(flow.dst_port());
+        Ok(())
+    }
```
Comment (drunkirishcoder) [inline framework/src/packets/tcp.rs]: not sure I like this function here. feels like unlike the changes to flow where it's pretty universal and will be used a lot. this feels very specific to that one example. I'd like not to add to the core api unless it's something we feel will get a lot of use.

File: framework/src/packets/udp.rs [hunk 8]
```diff
@@ -142,6 +142,15 @@ impl<E: IpPacket> Udp<E> {
...
     }
 
+    #[inline]
+    pub fn stamp_flow(&mut self, flow: Flow) -> Result<()> {
+        self.envelope_mut().set_src(flow.src_ip())?;
+        self.envelope_mut().set_dst(flow.dst_ip())?;
+        self.set_src_port(flow.src_port());
+        self.set_dst_port(flow.dst_port());
+        Ok(())
+    }
```
Comment (drunkirishcoder) [inline framework/src/packets/udp.rs]: same as the other comment.

File: examples/nat-tcp-v4/src/main.rs [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+
+    match PORT_MAP.read().unwrap().get(&flow) {
+        Some(s) => {
+            let _ = tcp.stamp_flow(*s);
+            tcp.cascade();
+        }
+        None => {
+            if *NEXT_PORT.get() < MAX_PORT {
+                let assigned_port = *NEXT_PORT.get();
+                NEXT_PORT.set(*NEXT_PORT.get() + 1);
```
[the remark this answers is not in the document]
Comment (zeeshanlakhani) [reply examples/nat-tcp-v4/src/main.rs]: @drunkirishcoder gotcha. But we still can use our atom for configuration etc? Agree to use atomic here.

File: examples/op-errors/src/main.rs (hunk 4 above)
Comment (zeeshanlakhani) [reply examples/op-errors/src/main.rs]: Good call

File: framework/src/config/mod.rs (hunk 5 above)
Comment (zeeshanlakhani) [reply framework/src/config/mod.rs]: Will do and agree

File: framework/src/packets/tcp.rs (hunk 7 above)
Comment (zeeshanlakhani) [reply framework/src/packets/tcp.rs]: Yep. I’ll just make specific ones in the nf

**Pull request merged**